sensor/kubernetes/complianceoperator/updater.go

```diff
@@ -53,9 +52,6 @@ type updaterImpl struct {
 }
 
 func (u *updaterImpl) Start() error {
-	if !features.ComplianceEnhancements.Enabled() {
-		return nil
-	}
```

Comment:
Just an idea: To avoid having this component (and the handler above) running if central does not have the capability but ensuring that it runs if central enables it (e.g. central upgrade), you could start/stop the `run` goroutine in the `Notify` function. When sensor goes online/offline every component should receive `common.SensorComponentEventCentralReachable`/`common.SensorComponentEventOfflineMode`.

Reply:
@lvalerom I would like to do something like that.  However, unless I'm missing something, it seems that Notify is called before we get the capabilities from central.  Am I missing something?  Is there another typical pattern in sensor to accomplish the same thing?

Reply:
AFAIK the only other place where we use the capabilities is during the sync process. We do not have a current pattern to achieve this but I think it would be a good (and simple 🤞) improvement to notify the components when the sync is done. I have opened a PR for it #8988.

Reply:
@dashrews78 Now Sensor components should receive `SensorComponentEventSyncFinished` in the Notify function.

Reply:
Sorry, I'm a little late to the party. I don't have a strong objection against `SensorComponentEventSyncFinished`, but it feels like we should've instead waited for the full handshake before sending `SensorComponentEventCentralReachable`? Sounds like a misleading event if by the time the components received the notification the connection with central is not _quite_ ready.